### PR TITLE
docs: add the development guide on how to run test

### LINKED ISSUE #1248: Provide a development.md or **Development Guide** in readme where you explain how to start project locally and do some basic manual tests and how to run existing tests in isolated docker env
As the title says, provide a `ReadMe` for developers in the repository so they know how to start projects locally, write tests, and run existing tests in isolated docker environment. Thus, it includes a couple of things: 
1. Guide to start project
2. How to add tests to the `test` directory
3. The docker commands to run tests

### PR DESCRIPTION
**Pull request by Gmin2**
**Description**

Added a guide on how to run test locally and in isolated docker enviorment.

**Related issue(s)**
Fixes #1248

### CODE REVIEW

**Review by derberg — COMMENTED** [review]
since we have a release process described in readme, don't you want dev guide to be there too? 

I think that for consistency, we should have both either in main readme, or both just mentioned in readme and details in separate markdown files

@lmgyuan thoughts on this one?

File: Development.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Development Guide
+
+This guide will help you set up the `generator` locally, run tests, and use Docker for isolated testing.
+
+## Getting Started
```
Comment (Florence-Njeri) [inline Development.md:5]:
```suggestion
## Getting started
```

File: Development.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Development Guide
```
Comment (Florence-Njeri) [inline Development.md:1]:
```suggestion
# Development guide
```

File: Development.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+cd generator
+```
+
+2. Install dependencies:
+
+```bash
+npm install
+```
+
+## Running Tests
```
Comment (Florence-Njeri) [inline Development.md:20]:
```suggestion
## Running tests
```

File: Development.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+### Local Testing
```
Comment (Florence-Njeri) [inline Development.md:22]:
```suggestion
### Local testing
```

File: Development.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+To run all tests locally:
+
+- Unit tests: `npm run generator:test:unit`
+- Integration tests: `npm run generator:test:integration`
+- CLI tests: `npm run generator:test:cli`
+
+### Adding Tests
```
Comment (Florence-Njeri) [inline Development.md:30]:
```suggestion
### Adding tests
```

File: Development.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+1. Create new test files in the appropriate directory under `test/`:
+
+2. Follow the existing test patterns.
+
+3. Run your new tests using the commands mentioned above.
+
+## Docker Isolated Testing
```
Comment (Florence-Njeri) [inline Development.md:38]:
```suggestion
## Docker isolated testing
```

File: Development.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+```bash
+docker run --rm -v ${PWD}:/app -w /app node:18 sh -c "
+cp -r /app /tmp/app &&
+cd /tmp/app &&
+npm install &&
+npm test
+"
+```
+
+This command does the following:
```
Comment (Florence-Njeri) [inline Development.md:55]:
```suggestion
This command above does the following:
```

File: Development.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+- Mounts the current directory to `/app` in the container
+- Copies the project to a temporary directory
+- Installs dependencies
+- Runs all tests
+
+Note: This approach ensures a clean environment for each test run by removing any existing `node_modules`.
+
+## Additional Commands
```
Comment (Florence-Njeri) [inline Development.md:63]:
```suggestion
## Additional commands
```

File: Development.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
+# Development guide
+
+This guide will help you set up the `generator` locally, run tests, and use Docker for isolated testing.
+
+## Getting started
```
Comment (derberg) [inline Development.md:5]: first person should fork the repo and setup the fork properly and configure it as described in https://github.com/asyncapi/community/blob/master/git-workflow.md

File: Development.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+To run all tests locally:
+
+- Unit tests: `npm run generator:test:unit`
+- Integration tests: `npm run generator:test:integration`
+- CLI tests: `npm run generator:test:cli`
+
+### Adding tests
+
+1. Create new test files in the appropriate directory under `test/`:
```
Comment (derberg) [inline Development.md:32]: test is no longer in root

**Review by derberg — COMMENTED** [review]
left 2 comments

what we are missing additionally is explanation on how to manually test templates features using react-template + you could fix template config in this PR, the version prop

**Comment by Gmin2** [comment]
> I think that for consistency, we should have both either in main readme, or both just mentioned in readme and details in separate markdown files
> 

@lmgyuan and i had a conversation and we both agreed to maake a seperate `developement.md`, i think we should also add the release process here only.

cc @derberg @lmgyuan

**Comment by Gmin2** [comment]
>  you could fix template config in this PR, the version prop

The connfig with the version of different templates right? , and if how do you want the content to be, like pointing to this docs (https://www.asyncapi.com/docs/tools/generator/versioning)

cc @derberg  @Florence-Njeri

File: Development.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+This guide will help you set up the `generator` locally, run tests, and use Docker for isolated testing.
+
+## Getting started
+
+1. Fork & Clone the repository:
+
+First fork the repository from github and then clone it,
+
+```bash
+git clone https://github.com/asyncapi/generator.git
```
Comment (derberg) [inline Development.md:12]:
```suggestion
git clone https://github.com/{your-username}/generator.git
```

File: Development.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+cd generator
```
Comment (derberg) [inline Development.md:13]:
```suggestion
cd apps/generator
```

File: Development.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+
+To run all tests locally:
+
+- Unit tests: `npm run generator:test:unit`
+- Integration tests: `npm run generator:test:integration`
+- CLI tests: `npm run generator:test:cli`
+
+### Adding tests
+
+1. Create new test files in the appropriate directory under `generator/apps/test/`:
```
Comment (derberg) [inline Development.md:36]:
```suggestion
1. Create new test files in the appropriate directory under `apps/generator/test/`:
```

File: Development.md [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+### Manually testing with test templates
+
+To test template features manually we have `react-template` and `nunjucks-template` in `apps/generator/test/templates`, you can use this templates to manually test your changes like this:
+
+1. Navigate to the generator directory:
+
+```bash
+cd apps/generator
+```
+2.Modify the react-template in `./test/test-templates/react-template` to test different features.
```
Comment (derberg) [inline Development.md:76]:
```suggestion
2. Modify the react-template in `./test/test-templates/react-template` to test different features.
```

**Review by derberg — COMMENTED** [review]
@Gmin2 ok then, please move release process part from readme to development.md

also please update https://github.com/asyncapi/generator/blob/master/apps/generator/test/test-templates/react-template/package.json#L10 so it is `apiVersion` and not `api`

File: Development.md (hunk 12 above)
Comment (Gmin2) [reply Development.md:13]: no it should be generator only as, git will make a new directory when cloning

Comment (derberg) [reply Development.md:13]: right, sorry

**Review by derberg — APPROVED** [review]

**Comment by Gmin2** [comment]
Think we can merge it 
@derberg @lmgyuan

**Pull request merged**